analyzers/src/SonarAnalyzer.Common/DiagnosticAnalyzer/SonarDiagnosticAnalyzer.cs

```diff
@@ -18,17 +18,36 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
 using Microsoft.CodeAnalysis.Diagnostics;
 
 namespace SonarAnalyzer.Helpers
 {
     public abstract class SonarDiagnosticAnalyzer : DiagnosticAnalyzer
     {
+        private const string IsParallelProcessing = "SONAR_DOTNET_ENABLE_PARALLEL_PROCESSING";
+        protected virtual bool ParallelProcessingDisabled => !TryGetBoolEnvironmentVariable(IsParallelProcessing, false);
+
+        protected abstract void Initialize(SonarAnalysisContext context);
+
         public sealed override void Initialize(AnalysisContext context)
         {
+            if (!ParallelProcessingDisabled)
+            {
+                context.EnableConcurrentExecution();
+            }
             Initialize(new SonarAnalysisContext(context, SupportedDiagnostics));
         }
 
-        protected abstract void Initialize(SonarAnalysisContext context);
+        private static bool TryGetBoolEnvironmentVariable(string envVar, bool defaultValue)
```

Comment:
Also, I would say that the parameters are not needed, the environment variable name is a constant in the class so it can be used directly, and the default value can be also defined as a constant or just used as a false in the method with a comment.